sqldelight-compiler/integration-tests/src/test/kotlin/com/example/PlayerQueries.kt

```diff
@@ -100,7 +94,18 @@ class PlayerQueries(private val queryWrapper: QueryWrapper, private val database
         team: String?,
         shoots: Shoots
     ) {
-        insertPlayer.execute(name, number, team, shoots)
+        val statement = database.prepareStatement(86, """
+                |INSERT INTO player
+                |VALUES (?1, ?2, ?3, ?4)
+                """.trimMargin(), SqlPreparedStatement.Type.INSERT, 4)
+        statement.bindString(1, name)
+        statement.bindLong(2, number)
+        statement.bindString(3, team)
+        statement.bindString(4, queryWrapper.playerAdapter.shootsAdapter.encode(shoots))
```

Comment:
yup makes sense. I'm also wondering since you always prepare, then bind, then execute in that order every time you want to run if the api should just be something like

```
fun SqlDatabase.executeQuery(
  identifier: Int,
  sql: () -> String,
  type: SqlPreparedStatement.Type,
  argumentCount: Int
  bind: Bindable.() -> Unit
): SqlCursor
```

and then the same for executing non cursor returning ones.

Reply:
since its generated/driver code only im not that worried about that for 1.0 tho